### PR TITLE
Fixed compound assignment

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/35870

### CODE REVIEW

File: src/Analyzers/Core/CodeFixes/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs [hunk 1]
```diff
@@ -43,6 +43,7 @@ internal abstract class AbstractUseCompoundAssignmentCodeFixProvider<
             TSyntaxKind assignmentOpKind, TExpressionSyntax left, SyntaxToken syntaxToken, TExpressionSyntax right);
         protected abstract TExpressionSyntax Increment(TExpressionSyntax left, bool postfix);
         protected abstract TExpressionSyntax Decrement(TExpressionSyntax left, bool postfix);
+        protected abstract SyntaxTriviaList PrepareRightExpressionLeadingTrivia(SyntaxTriviaList initialTrivia);
```
Comment (genlu) [inline src/Analyzers/Core/CodeFixes/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs:46]: nit: maybe just call this `RemoveLeadingWhitespaces` for clarity.

File: src/Analyzers/CSharp/Tests/UseCompoundAssignment/UseCompoundAssignmentTests.cs [hunk 2]
```diff
@@ -684,6 +684,28 @@ void M()
...
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestRightExpressionOnNextLine()
+        {
+            await TestInRegularAndScript1Async(
+@"public class C
+{
+    void M(int a)
+    {
+        a [||]= a +
+            10;
```
Comment (genlu) [inline src/Analyzers/CSharp/Tests/UseCompoundAssignment/UseCompoundAssignmentTests.cs:697]: Please add a test to cover leading newlines as well. thx

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Extensions/SyntaxTriviaExtensions.vb [hunk 3]
```diff
@@ -40,7 +40,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
             Return trivia.Kind = SyntaxKind.CommentTrivia OrElse trivia.Kind = SyntaxKind.DocumentationCommentTrivia
         End Function
 
-        <Extension()>
```
Comment (genlu) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Extensions/SyntaxTriviaExtensions.vb:43]: pls consider undo the change unrelated to the fix, or at least have them in a separate commit to keep history clean (although I have to admit we have been doing a terrible job on this ourselves :P)

File: src/Analyzers/Core/CodeFixes/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs (hunk 1 above)
Comment (DoctorKrolic) [reply src/Analyzers/Core/CodeFixes/UseCompoundAssignment/AbstractUseCompoundAssignmentCodeFixProvider.cs:46]: Not a good idea, considering the fact, that this method now also trims leading new lines for C#. More generic name is preffered here IMO

File: src/Analyzers/CSharp/CodeFixes/UseCompoundAssignment/CSharpUseCompoundAssignmentCodeFixProvider.cs [hunk 4]
```diff
@@ -47,6 +48,8 @@ private static ExpressionSyntax Postfix(SyntaxKind kind, ExpressionSyntax operan
         private static ExpressionSyntax Prefix(SyntaxKind kind, ExpressionSyntax operand)
             => SyntaxFactory.PrefixUnaryExpression(kind, operand);
 
+        protected override SyntaxTriviaList PrepareRightExpressionLeadingTrivia(SyntaxTriviaList initialTrivia) => initialTrivia.SkipWhile(el => el.Kind() is SyntaxKind.WhitespaceTrivia or SyntaxKind.EndOfLineTrivia).ToSyntaxTriviaList();
```
Comment (genlu) [inline src/Analyzers/CSharp/CodeFixes/UseCompoundAssignment/CSharpUseCompoundAssignmentCodeFixProvider.cs:51]: You could use this [`IsKind()`](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/J/s/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs/SyntaxNodeExtensions.cs,70) overload instead

Comment (DoctorKrolic) [reply src/Analyzers/CSharp/CodeFixes/UseCompoundAssignment/CSharpUseCompoundAssignmentCodeFixProvider.cs:51]: There is only 1 overload for `SyntaxTrivia`, wich is a type of `el` here, that takes only 1 `SyntaxKind`. No overloads with 2 `SyntaxKind`s for comparing exist.

Comment (genlu) [reply src/Analyzers/CSharp/CodeFixes/UseCompoundAssignment/CSharpUseCompoundAssignmentCodeFixProvider.cs:51]: Oops, my bad. [This ](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/SyntaxTriviaExtensions.cs,24)is what I meant to say. And I just noticed there's already an [`IsWhitespaceOrEndOfLine`]([IsWhitespaceOrEndOfLine](https://sourceroslyn.io/Microsoft.CodeAnalysis.CSharp.Workspaces/R/07abdeb1515ceffd.html)) helper

**Review by genlu — APPROVED** [review]
Thanks!
